mock/py/mockbuild/shadow_utils.py

```diff
@@ -15,14 +15,18 @@ def __init__(self, root):
         self.root = root
 
     @property
-    def _chroot_opts(self):
-        return ["--prefix", self.root.make_chroot_path()]
+    def chroot_dir(self):
+        return self.root.make_chroot_path()
 
     def _execute_command(self, command, can_fail=False):
         with self.root.uid_manager.elevated_privileges():
-            # Execute the command _on host_, not in bootstrap (where we're not
-            # sure how old shadow-utils are).
-            do_with_status(command + self._chroot_opts, raiseExc=not can_fail)
+            # Execute the command in bootstrap, since host configuration may not permit
+            # ordinary user addition, such as would happen with freeipa-provided subids
+            if ('use_chroot_shadow_utils' in self.root.config) and self.root.config['use_chroot_shadow_utils']:
+                do_with_status(command, chrootPath=self.chroot_dir, raiseExc=not can_fail)
```

Comment:
I think you are not executing in the bootstrap chroot, but in buildroot (build chroot) (see doOutChroot).

Reply:
Odd. This worked like I expected it to. I'll investigate it further before finalizing.